compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -348,6 +360,64 @@ void RecordMinMax::profileRawData(const std::string &mode, const std::string &in
   update_quantparam(getObserver(), mode, min_percentile, max_percentile);
 }
 
+BufferInputData RecordMinMax::importH5Data(const std::string &input_data_path)
+{
+  try
+  {
+    dio::hdf5::HDF5Importer importer(input_data_path);
+    importer.importGroup("value");
+
+    bool is_raw_data = importer.isRawData();
+
+    const auto num_records = importer.numData();
+    if (num_records == 0)
+      throw std::runtime_error("The input data file does not contain any record.");
+
+    const auto input_nodes = loco::input_nodes(_module->graph());
+    const auto num_inputs = input_nodes.size();
+
+    BufferInputData vector_input_data(num_records);
+
+    // Read inputs to vector_input_data
+    for (int i = 0; i < num_records; ++i)
+    {
+      if (num_inputs != static_cast<uint32_t>(importer.numInputs(i)))
+        throw std::runtime_error("Wrong number of inputs.");
+
+      for (uint32_t input_idx = 0; input_idx < num_inputs; input_idx++)
+      {
+        const auto *input_node = loco::must_cast<const luci::CircleInput *>(input_nodes[input_idx]);
+        assert(input_node->index() == input_idx);
+        checkInputDimension(input_node);
+        std::vector<char> input_data(getTensorSize(input_node));
```

Comment:
As @jinevening commented, can you use `std::vector<char>` as a meaning for this?

And revise `using BufferInputData = std::vector<std::vector<std::vector<char>>>;`
too.

Reply:
Done